rust/stracciatella/src/vfs/android.rs

```diff
@@ -57,6 +65,7 @@ impl AssetManagerFs {
         Ok(Rc::new(AssetManagerFs {
             base_path: base_path.to_owned(),
             asset_manager,
+            canonicalization_cache: Mutex::new(LruCache::new(CANONICALIZATION_CACHE_SIZE)),
```

Comment:
What does the Mutex protects us from? If there are concurrent access (do we really do so?) to the same key, isn't the worst case just false cache-miss?

Can we simplify the code and tolerate race conditions?

Reply:
It's hard to simplify this in rust, as there are strict interior mutablility rules. We could use some lock-free cache instead of `lru`, I just opted with lru because it is the more widely used choice. Or I could try to use `Cell` or `RefCell` to move ownership checks to runtime.

Reply:
I see. Using lock-free sounds good, but I don't think we really need that.  Let's just keep what we have.